src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let lastResult = initialValue;
+    let i = 0;
+
+    if (arguments.length < 2) {
```

Comment:
I don't know which way is better whether to check initialValue for equality to undefined or to check arguments length. Or maybe it's not an issue at all. I hope mentors can clarify it.

Reply:
Anyway, good job!

Reply:
This method also works, but it is definitely better to check if `initialValue === undefined`. This shows the exact logic of your code.